packages/protocol/contracts/tokenvault/ERC721Vault.sol

```diff
@@ -106,7 +106,7 @@ contract ERC721Vault is BaseNFTVault, IERC721Receiver, IERC165Upgradeable {
             if (ctoken.chainId == block.chainid) {
                 token = ctoken.addr;
                 for (uint256 i; i < tokenIds.length; ++i) {
-                    ERC721Upgradeable(token).transferFrom({
+                    ERC721Upgradeable(token).safeTransferFrom({
```

Comment:
could you also change "BridgedERC721.sol"  from
`return ERC721Upgradeable.transferFrom(from, to, tokenId);` to 
`return ERC721Upgradeable.safeTransferFrom(from, to, tokenId);`

Reply:
There is another `.transferFrom` call in this file, please also replace it with 'safeTransferFrom'

Reply:
The other `.transferFrom` in this file sends the `ERC721` to `address(this)` (this contract) which already implements `onERC721received`. So there is no need to use `.safeTransferFrom`. Using `.transferFrom` will optimize a bit more gas.

Reply:
for "BridgedERC721.sol", I'd recommend overriding `.safeTransferFrom` to call `ERC721Upgradeable.safeTransferFrom(from, to, tokenId);` and use this function elsewhere rather than implementing `.safeTransferFrom` in the `.transferFrom`.